### PR TITLE
dev: replace starknet_crypto::FieldElement with starknet-types-core 

### PR DESCRIPTION
**Pull request by eugypalu**
Resolves: #1233 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?
Replace starknet_crypto::FieldElement with starknet-types-core 

# Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by tcoratger** [comment]
Blocked by https://github.com/dojoengine/dojo/issues/2177

**Comment by tcoratger** [comment]
@eugypalu Should be good now, let us see if there are issues during the ci, but at least all the replacements should be done :)

File: src/eth_provider/provider.rs [hunk 1]
```diff
@@ -847,9 +853,12 @@ where
...
                 .await
                 .map_err(|_| SignatureError::SigningFailure)?;
-            self.starknet_provider.add_invoke_transaction(tx).await.map_err(KakarotError::from)?;
+            self.starknet_provider
+                .add_invoke_transaction(BroadcastedInvokeTransaction::V1(tx))
+                .await
+                .map_err(KakarotError::from)?;
 
-            *nonce += 1u8.into();
+            *nonce += Felt::from(1);
```
Comment (greged93) [inline src/eth_provider/provider.rs:861]:
```suggestion
            *nonce += Felt::ONE;
```

File: src/test_utils/hive/mod.rs [hunk 2]
```diff
@@ -99,12 +102,12 @@ impl HiveGenesisConfig {
 
                 let key = get_storage_var_address("ERC20_allowances", &[starknet_address, kakarot_address])?;
                 fee_token_storage.insert(key, u128::MAX.into());
-                fee_token_storage.insert(key + 1u8.into(), u128::MAX.into());
+                fee_token_storage.insert(key + Felt::from(1), u128::MAX.into());
```
Comment (greged93) [inline src/test_utils/hive/mod.rs:105]:
```suggestion
                fee_token_storage.insert(key + Felt::ONE, u128::MAX.into());
```

File: src/test_utils/hive/mod.rs [hunk 3]
```diff
@@ -168,21 +169,19 @@ mod tests {
...
             assert_eq!(contract.balance, Some(account.balance));
             // Check the storage
             for (key, value) in account.storage.unwrap_or_default() {
-                let key = get_storage_var_address(ACCOUNT_STORAGE, &split_u256::<FieldElement>(key)).unwrap();
+                let key = get_storage_var_address(ACCOUNT_STORAGE, &split_u256::<Felt>(key)).unwrap();
                 let low =
                     U256::from_be_slice(contract.storage.as_ref().unwrap().get(&key).unwrap().to_bytes_be().as_slice());
                 let high = U256::from_be_slice(
-                    contract.storage.as_ref().unwrap().get(&(key + 1u8.into())).unwrap().to_bytes_be().as_slice(),
+                    contract.storage.as_ref().unwrap().get(&(key + Felt::from(1))).unwrap().to_bytes_be().as_slice(),
```
Comment (greged93) [inline src/test_utils/hive/mod.rs:184]:
```suggestion
                    contract.storage.as_ref().unwrap().get(&(key + Felt::ONE)).unwrap().to_bytes_be().as_slice(),
```

File: src/test_utils/katana/genesis.rs [hunk 4]
```diff
@@ -259,7 +259,7 @@ impl KatanaGenesisBuilder<Initialized> {
 
         // Set the allowance for the EOA to the Kakarot contract.
         let key = get_storage_var_address("ERC20_allowances", &[*starknet_address, kakarot_address])?;
-        let storage = [(key, u128::MAX.into()), (key + 1u8.into(), u128::MAX.into())].into_iter();
+        let storage = [(key, u128::MAX.into()), (key + Felt::from(1), u128::MAX.into())].into_iter();
```
Comment (greged93) [inline src/test_utils/katana/genesis.rs:262]:
```suggestion
        let storage = [(key, u128::MAX.into()), (key + Felt::ONE, u128::MAX.into())].into_iter();
```

File: src/test_utils/katana/genesis.rs [hunk 5]
```diff
@@ -283,7 +283,7 @@ impl KatanaGenesisBuilder<Initialized> {
         let key = get_storage_var_address("ERC20_balances", &[*starknet_address])?;
         let amount_split = split_u256::<u128>(amount);
 
-        let storage = [(key, amount_split[0].into()), (key + 1u8.into(), amount_split[1].into())].into_iter();
+        let storage = [(key, amount_split[0].into()), (key + Felt::from(1), amount_split[1].into())].into_iter();
```
Comment (greged93) [inline src/test_utils/katana/genesis.rs:286]:
```suggestion
        let storage = [(key, amount_split[0].into()), (key + Felt::ONE, amount_split[1].into())].into_iter();
```

**Review by greged93 — COMMENTED** [review]
small nits otherwise looks good

**Review by greged93 — APPROVED** [review]

**Pull request merged**